Lib/test/test_ntpath.py

```diff
@@ -321,6 +322,12 @@ def test_realpath_basic(self):
         self.assertPathEqual(ntpath.realpath(os.fsencode(ABSTFN + "1")),
                          os.fsencode(ABSTFN))
 
+        # gh-88013: call ntpath.realpath with binary drive name may raise a
+        # TypeError. The drive should not exist to reproduce the bug.
+        drives = {f"{c}:\\" for c in string.ascii_uppercase} - set(os.listdrives())
+        d = drives.pop().encode()
```

Comment:
```suggestion
        for c in string.ascii_uppercase:
            d = f"{c}:\\"
            if not ntpath.exists(d):
                break
        else:
            raise OSError("No free drive letters available")
```

Longer but perhaps simpler to follow?

Reply:
LGTM!